File: src/jray/common/Matrix3.java
package jray.common;

public class Matrix3 {
	public double[] data;
	
	public Matrix3(double[] data){
		this.data = data;
	}
	
	public Matrix3(){
		data = new double[9];
	}
	
	public static void vmm(Vect3 v, Matrix3 m, Vect3 erg){
		erg.data[0] = v.data[0]*m.data[0] + v.data[1]*m.data[3] + v.data[2]*m.data[6];
		erg.data[1] = v.data[0]*m.data[1] + v.data[1]*m.data[4] + v.data[2]*m.data[7];
		erg.data[2] = v.data[0]*m.data[2] + v.data[1]*m.data[5] + v.data[2]*m.data[8];
	}
}

File: JRayXLib/JRayXLib/Shapes/Matrix3.cs


namespace JRayXLib.Shapes
{
    public class Matrix3 {
        public double[] Data;
	
        public Matrix3(double[] data){
            Data = data;
        }
	
        public Matrix3(){
            Data = new double[9];
        }
	
        public static void VectorMatrixProduct(Vect3 v, Matrix3 m, ref Vect3 erg){
            erg.Data[0] = v.Data[0]*m.Data[0] + v.Data[1]*m.Data[3] + v.Data[2]*m.Data[6];
            erg.Data[1] = v.Data[0]*m.Data[1] + v.Data[1]*m.Data[4] + v.Data[2]*m.Data[7];
            erg.Data[2] = v.Data[0]*m.Data[2] + v.Data[1]*m.Data[5] + v.Data[2]*m.Data[8];
        }
    }
}
